src/client/components/Inputs/TextArea/TextArea.tsx

```diff
@@ -1,31 +1,38 @@
 import './TextArea.scss'
-import React, { forwardRef, TextareaHTMLAttributes, useEffect, useImperativeHandle, useRef } from 'react'
+import React, { forwardRef, TextareaHTMLAttributes, useImperativeHandle, useRef } from 'react'
 
-type Props = Pick<TextareaHTMLAttributes<HTMLTextAreaElement>, 'disabled' | 'onChange' | 'onPaste' | 'value'>
+import useResize from './hooks/useResize'
+
+type Props = Pick<
+  TextareaHTMLAttributes<HTMLTextAreaElement>,
+  'disabled' | 'onChange' | 'onPaste' | 'placeholder' | 'rows' | 'value'
+> & { maxHeight?: number | null }
 
 const TextArea = forwardRef<HTMLTextAreaElement, Props>((props, outerRef) => {
-  const { disabled, onChange, onPaste, value } = props
+  const { disabled, maxHeight, onChange, onPaste, placeholder, rows, value } = props
 
   const textAreaRef = useRef<HTMLTextAreaElement>(null)
   useImperativeHandle(outerRef, () => textAreaRef.current!, [])
 
-  useEffect(() => {
-    const textArea = textAreaRef.current
-    textArea.style.height = 'auto'
-    textArea.style.height = `${textArea.scrollHeight}px`
-  }, [disabled, textAreaRef, value])
+  useResize(textAreaRef, maxHeight, value)
 
   return (
     <textarea
       className="textarea validation-error-sensitive-field"
       disabled={disabled}
       onChange={onChange}
       onPaste={onPaste}
+      placeholder={placeholder}
       ref={textAreaRef}
-      rows={1}
+      rows={rows}
       value={value}
     />
   )
 })
 
+TextArea.defaultProps = {
+  maxHeight: null,
+  rows: 1,
+} as Props
```

Comment:
If I don't add it I get `error    defaultProp "rows" has no corresponding propTypes declaration  react/default-props-match-prop-types` in that line. I think this is because eslint doesn't pick up that rows gets its type declaration from 
```
Pick<
  TextareaHTMLAttributes<HTMLTextAreaElement>,
  'disabled' | 'onChange' | 'onPaste' | 'placeholder' | 'rows' | 'value'
>
```
(If I add a default value for `placeholder`I get the same error)